Volume/Views/WeeklyDebriefView.swift

```diff
@@ -61,62 +85,149 @@ struct WeeklyDebriefView: View {
-                            Text("Continue to Volume")
-                                .foregroundColor(.volume.orange)
-                                .font(.helveticaBold(size: 16))
-                        }
-                        .frame(width: 184, height: 45, alignment: .center)
-                        .overlay(RoundedRectangle(cornerRadius: 26)
-                        .stroke(Color.volume.orange, lineWidth: 2))
-                        .padding(.top, 200)
-                        Spacer()
+                        Image.volume.logo
+                            .resizable()
+                            .frame(width: 245, height: 75)
+                        Text("Stay updated with Cornell student publications, all in one place")
+                            .font(.newYorkRegular(size: 16))
+                    }
+                    .padding(.top, 200)
+                    Button {
+                        isOpen = false
+                    } label: {
+                        Text("Continue to Volume")
+                            .foregroundColor(.volume.orange)
+                            .font(.helveticaBold(size: 16))
                     }
                 }
-                .padding([.leading, .trailing], 50)
+                .padding(.horizontal, 50)
             }
             .tabViewStyle(PageTabViewStyle())
         }
-        .onAppear() {
+    }
+    
+    var body: some View {
+        VStack {
+            switch state {
+            case .loading:
+                BigReadArticleRow.Skeleton() // TODO: This is until we have a SkeletonView for WeeklyDebrief
+            case .reloading(let debriefDisplayInfo):
+                makeView(debriefDisplayInfo: debriefDisplayInfo)
+            case .results(let debriefDisplayInfo):
+                makeView(debriefDisplayInfo: debriefDisplayInfo)
+            }
+        }
+        .onAppear {
             UIPageControl.appearance().currentPageIndicatorTintColor = .gray
             UIPageControl.appearance().pageIndicatorTintColor = .lightGray
+            fetchInfoFor(debrief: weeklyDebrief)
         }
-        .onDisappear() {
+        .onDisappear {
             UIPageControl.appearance().currentPageIndicatorTintColor = nil
             UIPageControl.appearance().pageIndicatorTintColor = nil
         }
     }
+    
+    // MARK: - Copying from BrowserView
+    
+    private func asyncPopulateDisplayInfo(recievedArticles: Int = 0, ids: [ArticleID], readNotRandom: Bool, displayInfo: WeeklyDebriefDisplayInfo, completion: @escaping (WeeklyDebriefDisplayInfo) -> ()) {
+        if recievedArticles == ids.count {
+            completion(displayInfo)
+            return
+        }
+        
+        cancellableArticleQuery = Network.shared.publisher(for: GetArticleByIdQuery(id: ids[recievedArticles]))
+            .sink { completion in
+                if case let .failure(error) = completion {
+                    print(error.localizedDescription)
```

Comment:
follow the standard error printing format that han established for volume

you can see at [this PR](https://github.com/cuappdev/volume-ios/pull/79)